crossdock/server/tchannel.go

```diff
@@ -73,7 +72,11 @@ func (s *Server) callDownstreamTChannel(ctx context.Context, downstream *tracete
 
 	println("calling downstream over tchannel")
 
-	thriftClient := thrift.NewClient(s.channel, "go", nil)
+	// These two operations are idempotent
+	subchannel := s.channel.GetSubChannel(downstream.Host, tchannel.Isolated)
+	subchannel.Peers().Add(s.channel.PeerInfo().HostPort)
```

Comment:
Also, I would prefer if we check that there are currently either no peers on this sub-channel, or there is just one peer with exactly the same host:port. Otherwise we might add a different peer and end up introducing a hard to find but that can give false positives in the tests.